### ISSUE TITLE
MultiSelect: onBlur event mismatch

### ISSUE THREAD

**Issue by domino2** [primefaces/primereact#2833]
### Describe the bug

The component's event "onBlur" is fired on, not only when focus is lost.

### Reproducer

https://codesandbox.io/s/objective-wind-13w45j?file=/src/App.jsx:448-459

### PrimeReact version

7.2.1

### React version

17.x

### Language

ALL

### Build / Runtime

Create React App (CRA)

### Steps to reproduce the behavior

1. click on the `MultiSelect` to show the list of options
2. pick any option
3. `onBlur` event will increase counter above 

1. click on the `MultiSelect` to show the list of options and then click on filter input or "select all" checkbox
2. `onBlur` event will increase counter above 

### Expected behavior

The `onBlur` event should to be triggered only when then component as a whole will lost the focus.

So it shouldn't be triggered during options selection nor filtering, but only when user will click (on|to) some(thing|where) else.

**Comment by melloware** [comment]
I fixed this by not calling onBlur if the Overlay is showing.  This seems to fix the issue and blur is only called now when you leave the field with the overlay closed.

**Comment by JosuerBague** [comment]
Encountered similar behaviour today, but onBlur only gets called on the initial option selection after the panel first shows.

**Comment by JosuerBague** [comment]
Oh I guess it gets triggered, because the focus switches from the multiSelect input to the multiSelect option thus triggering the onBlur event.

**Comment by melloware** [comment]
Yes as part of ARIA WCAG accesibility that is what accessibility says should happen @JosuerBague .

**Comment by AndrewBoklashko** [comment]
Hello folks
Trying to understand if this `onBlur` behavior is expected(e.g. recommended by ARIA WCAG) or is this something that should be fixed?

I see a similar issue with the `Calendar` component triggering `onBlur` when calendar icon is clicked. There is a workaround, so it's not critical:
```
const calendarRef = useRef<Calendar>(null);
const handleBlur = useCallback(
  (e: FocusEvent<HTMLInputElement>) => {
    if (calendarRef.current.getElement().contains(e.relatedTarget)) {
      // calendar icon click -> do nothing
      return;
    }

    onBlur?.(e);
  },
  [onBlur]
);
```

My question: is this behavior generally expected when working with all PrimeReact components? Or will it eventually change to only blur when whole component loses focus as requested in the issue description?

**Comment by melloware** [comment]
@AndrewBoklashko a PR would be welcome!
